### ISSUE TITLE
V2X

### ISSUE THREAD

**Issue by ZZG5220** [daohu527/dig-into-apollo#41]
看了几天，还没搞明白V2X模块的输入和输出是什么呢？官方开发文档中有上机教程，感知，融合，规划等多个模块都有，V2X怎么来上机调试呢？（大佬，问的问题有点小儿科，还希望大佬解答下）

**Comment by daohu527** [comment]
The V2X module is used to communicate with v2x devices such as `RSU`. You can refer to the v2x specification, which defines the format and purpose of this information.

This tutorial has some [introductions ](https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x)to v2x, debugging needs to start the v2x module and send v2x messages

**Comment by ZZG5220** [comment]
看了这个文档解析https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x，还是不太明白，主要是不明白这个模块的输入和输出，不明白和其他模块之间的关系，更不明白如何使用。看OBU，OS接口还涉及到grpc，就更不明白了。

**Comment by pjs9115916** [comment]
V2X是需要路侧设施的，可以粗爆理解为路侧有个服务器，车端为一个客户端，车和路侧设施进行数据交互。举个例子，红绿灯信息可以通过路侧设施直接发送红绿灯状态和倒计时给车辆，那路侧怎么知道发给谁呢，一个可能的情况，车上传自己的精确位置和前进方向。这个例子就包含了输入输出。发自我的荣耀手机-------- 原始邮件 --------发件人： ZZG5220 ***@***.***>日期： 2021年8月3日周二 晚上8:53收件人： daohu527/dig-into-apollo ***@***.***>抄送： Subscribed ***@***.***>主    题： Re: [daohu527/dig-into-apollo] V2X (#41)

看了这个文档解析https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x，还是不太明白，主要是不明白这个模块的输入和输出，不明白和其他模块之间的关系，更不明白如何使用。看OBU，OS接口还涉及到grpc，就更不明白了。

**Comment by daohu527** [comment]
Yes, you must first understand the business scenario, and then look at the code, otherwise it may find it very difficult

**Comment by ZZG5220** [comment]
> 
> V2X是需要路侧设施的，可以粗爆理解为路侧有个服务器，车端为一个客户端，车和路侧设施进行数据交互。举个例子，红绿灯信息可以通过路侧设施直接发送红绿灯状态和倒计时给车辆，那路侧怎么知道发给谁呢，一个可能的情况，车上传自己的精确位置和前进方向。这个例子就包含了输入输出。发自我的荣耀手机-------- 原始邮件 --------发件人： ZZG5220 ***@***.***>日期： 2021年8月3日周二 晚上8:53收件人： daohu527/dig-into-apollo ***@***.***>抄送： Subscribed ***@***.***>主 题： Re: [daohu527/dig-into-apollo] V2X (#41) 看了这个文档解析[https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x，还是不太明白，主要是不明白这个模块的输入和输出，不明白和其他模块之间的关系，更不明白如何使用。看OBU，OS接口还涉及到grpc，就更不明白了。](https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x%EF%BC%8C%E8%BF%98%E6%98%AF%E4%B8%8D%E5%A4%AA%E6%98%8E%E7%99%BD%EF%BC%8C%E4%B8%BB%E8%A6%81%E6%98%AF%E4%B8%8D%E6%98%8E%E7%99%BD%E8%BF%99%E4%B8%AA%E6%A8%A1%E5%9D%97%E7%9A%84%E8%BE%93%E5%85%A5%E5%92%8C%E8%BE%93%E5%87%BA%EF%BC%8C%E4%B8%8D%E6%98%8E%E7%99%BD%E5%92%8C%E5%85%B6%E4%BB%96%E6%A8%A1%E5%9D%97%E4%B9%8B%E9%97%B4%E7%9A%84%E5%85%B3%E7%B3%BB%EF%BC%8C%E6%9B%B4%E4%B8%8D%E6%98%8E%E7%99%BD%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8%E3%80%82%E7%9C%8BOBU%EF%BC%8COS%E6%8E%A5%E5%8F%A3%E8%BF%98%E6%B6%89%E5%8F%8A%E5%88%B0grpc%EF%BC%8C%E5%B0%B1%E6%9B%B4%E4%B8%8D%E6%98%8E%E7%99%BD%E4%BA%86%E3%80%82) —You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

好像有点明白了，能冒昧问下，有比较清晰的解读apollo中v2x模块的文档或者代码解析的资料嘛，不胜感激！

**Comment by pjs9115916** [comment]
你要看v2x就去看看车路协同的标准，没必要揪着Apollo的代码发自我的荣耀手机-------- 原始邮件 --------发件人： ZZG5220 ***@***.***>日期： 2021年8月3日周二 晚上9:28收件人： daohu527/dig-into-apollo ***@***.***>抄送： pjs9115916 ***@***.***>, Comment ***@***.***>主    题： Re: [daohu527/dig-into-apollo] V2X (#41)

V2X是需要路侧设施的，可以粗爆理解为路侧有个服务器，车端为一个客户端，车和路侧设施进行数据交互。举个例子，红绿灯信息可以通过路侧设施直接发送红绿灯状态和倒计时给车辆，那路侧怎么知道发给谁呢，一个可能的情况，车上传自己的精确位置和前进方向。这个例子就包含了输入输出。发自我的荣耀手机-------- 原始邮件 --------发件人： ZZG5220 @.>日期： 2021年8月3日周二 晚上8:53收件人： daohu527/dig-into-apollo @.>抄送： Subscribed @.***>主 题： Re: [daohu527/dig-into-apollo] V2X (#41) 看了这个文档解析https://github.com/daohu527/dig-into-apollo/tree/main/modules/v2x，还是不太明白，主要是不明白这个模块的输入和输出，不明白和其他模块之间的关系，更不明白如何使用。看OBU，OS接口还涉及到grpc，就更不明白了。 —You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

好像有点明白了，能冒昧问下，有比较清晰的解读apollo中v2x模块的文档或者代码解析的资料嘛，不胜感激！

**Comment by daohu527** [comment]
Here is a [doc](http://www.caict.ac.cn/kxyj/qwfb/bps/201806/P020180621609834833905.pdf) about v2x, You can also view more documents on the Internet
